File: tests/wasm2lang_09_crc32.harness.java
{
    WasmModule mod = new WasmModule(new java.util.LinkedHashMap<>(), memBuffer);
    mod.alignHeapTop();

    java.util.Map<String, Object> _data = w2lLoadSharedData(System.getProperty("w2l.testname", ""));
    @SuppressWarnings("unchecked")
    java.util.List<String> inputs = (java.util.List<String>) _data.get("crc32_inputs");
    int scratch = 0;

    for (String str : inputs) {
        byte[] bytes = str.getBytes(java.nio.charset.StandardCharsets.US_ASCII);
        for (int i = 0; i < bytes.length; i++) {
            memBuffer.put(scratch + i, bytes[i]);
        }
        mod.exerciseCrc32(scratch, bytes.length);
    }

    w2lDumpCRC(memBuffer);
}

/exit

File: tests/wasm2lang_09_crc32.harness.mjs
'use strict';

const moduleImports = {};

const runTest = function (buff, out, exports, data) {
  exports.alignHeapTop();
  const scratch = 0;
  const view = new Uint8Array(buff);

  for (const str of data.crc32_inputs) {
    for (let i = 0; i < str.length; i++) {
      view[scratch + i] = str.charCodeAt(i);
    }
    exports.exerciseCrc32(scratch, str.length);
  }
};

const dumpMemory = true;

export { dumpMemory, moduleImports, runTest };
